### ISSUE TITLE
Mailprocessing: Reply-To header are ignored

### ISSUE THREAD

**Issue by hanneshal** [zammad/zammad#203]
If I send a mail via a generic form mailer like this:

```
X-PHP-Originating-Script: 33:form-handler.php
From: foobar@bar.com
MIME-Version: 1.0
Content-type: text/plain; charset=utf-8
Content-Transfer-Encoding: quoted-printable
Reply-To: hannes@myspecialdomain.com
Message-Id: <20160919130314.34EEB1DA1B2@bar.com>
Date: Mon, 19 Sep 2016 15:03:14 +0200 (CEST)
X-znuny-MailScanner-Information: Please contact the ISP for more information
X-znuny-MailScanner-ID: 5E64620C3DE3.A1C7C
X-znuny-MailScanner: Found to be clean
X-znuny-MailScanner-From: www-data@bar.com
X-Spam-Status: No
```

The Reply-To header, which is this customer address, is ignored.

**Comment by olivergrahl** [comment]
Any objections against [this change](https://github.com/olivergrahl/zammad/commit/d7b35a9367379596be7388fc9eb0f949add044c1) to ignore the From header in favour of the Reply-To header? Would be happy to send a PR.

**Comment by thorsteneckel** [comment]
Hi @olivergrahl - I hadn't time yet to take a deeper look but looks promising. Just one question so far: Could you please use a more speaking variable name instead of `from_from`? Maybe something like `reply_to`?

**Comment by olivergrahl** [comment]
Okay, `from_from` was probably neither creative nor very self explanatory. :-)  I renamed it to `from_full` which is possibly not very self explanatory either. `from_full` holds the complete address with name and email, while `from_email`, `from_domain`, `from_display_name` hold its parsed sub elements. All of them are based on either `reply-to`, `from` or `return-path`. Let me know if you come up with a better name.

**Comment by olivergrahl** [comment]
Let me add that this code works well here, but the GUI displays the "Reply All" button, which would reply to the `reply-to` (VON below) and the `to` address (AN below, the email channel inbound address), and I believe that doesn't make too much sense. I'm not sure how this could be prevented, though. 
![grafik](https://cloud.githubusercontent.com/assets/4621219/21815285/57606fda-d75c-11e6-8e6d-4beb21e3a8de.png)

**Comment by rolfschmidt** [comment]
Hi all,
we decided to create an own column to save the reply to information. It will be displayed now in the detail information of the article:

![grafik](https://cloud.githubusercontent.com/assets/5445062/25440869/fd519ce2-2aa0-11e7-9fd6-bb81556bbc5b.png)

In case of an email reply it will used as recipient:

![grafik](https://cloud.githubusercontent.com/assets/5445062/25440914/1580cf40-2aa1-11e7-9651-0e123ffff4ea.png)

@olivergrahl Thank you very much for your support and your initiative pull request 👍 I hope this solution will fix your problem case too. If not you can gladly reopen the issue with more information about your case.

The bugfix is now available in the develop branch.

-Rolf

**Comment by olivergrahl** [comment]
Sounds like a solid solution, thanks!! 
(BTW, I did not initiate that PR, as I knew my solution wasn't that clever. Looks like someone else did...)

**Comment by rolfschmidt** [comment]
I'm sorry. @chschenk Please see https://github.com/zammad/zammad/issues/203#issuecomment-297435934

**Comment by fthommen** [comment]
Any chance, that this fix is (back)ported to Zammad 1.5?  The ignorance of the Reply-To header leads to at least 10 wrongly routed/"disappearing" mails per day and zammad 2.x is not out yet...
